### PR TITLE
Neptune system test tweaks

### PR DESCRIPTION
**Pull request by ferruzzi**
Some small adjustments were needed to make this run for our dashboard.

- System Tests should be self-contained; rather than accepting a cluster-id passed in, the test now creates and tears down the cluster
- Only one of our system tests uses `pendulum.datetime`, the rest all use `datetime.datetime`
- Trivial adjustments to the DAG parameters to match the other tests (added an "example" tag and set the same start date) 
- Our system test runners do not yet support deferrable mode, removed that for now

Local run results: 
image

(tagging @ellisms since the test is his contribution)

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by o-nikolas — COMMENTED** [review]
LGTM, can you run this locally and share a passing result?

**Comment by ferruzzi** [comment]
> LGTM, can you run this locally and share a passing result?

@o-nikolas  - Updated PR description with a screenshot of the local run success.

File: tests/system/providers/amazon/aws/example_neptune.py [hunk 1]
```diff
@@ -16,45 +16,66 @@
...
+sys_test_context_task = SystemTestContextBuilder().build()
 
-with DAG(DAG_ID, schedule="@once", start_date=pendulum.datetime(2024, 1, 1, tz="UTC"), catchup=False) as dag:
+
+@task
+def create_cluster(cluster_id):
+    hook = NeptuneHook()
+    hook.conn.create_db_cluster(DBClusterIdentifier=cluster_id, Engine="neptune", DeletionProtection=False)
+
+    hook.wait_for_cluster_availability(cluster_id=cluster_id)
```
Comment (o-nikolas) [inline tests/system/providers/amazon/aws/example_neptune.py]: Tiniest of nits:
 ```suggestion
    hook.conn.create_db_cluster(DBClusterIdentifier=cluster_id, Engine="neptune", DeletionProtection=False)
    hook.wait_for_cluster_availability(cluster_id=cluster_id)
```

**Review by o-nikolas — APPROVED** [review]

**Review by syedahsn — APPROVED** [review]
LGTM

**Review by eladkal — APPROVED** [review]

**Pull request merged**